### PR TITLE
fix volume covert missing update accessibility

### PR DESCRIPTION
**Pull request by Vacant2333**
#### What this PR does / why we need it:
fix issue:https://github.com/hwameistor/hwameistor/issues/933
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/local-storage/member/controller/volume_convert_task_worker.go [hunk 1]
```diff
@@ -310,3 +347,25 @@ func (m *manager) volumeConvertCleanup(convert *apisv1alpha1.LocalVolumeConvert)
...
+		logCtx.WithError(err).Error("Failed to query volume", volumeName)
+		return nil, err
+	}
+	return vol, nil
+}
+
+func (m *manager) queryLocalVolumeGroup(volumeGroupName string, ctx context.Context, logCtx *log.Entry) (*apisv1alpha1.LocalVolumeGroup, error) {
+	lvg := &apisv1alpha1.LocalVolumeGroup{}
+	if err := m.apiClient.Get(ctx, types.NamespacedName{Name: volumeGroupName}, lvg); err != nil {
+		if !errors.IsNotFound(err) {
```
Comment (SSmallMonster) [inline pkg/local-storage/member/controller/volume_convert_task_worker.go:363]: why handle `NotFoundError` here. May return error directly and leave it handled at where it called.

**Comment by SSmallMonster** [comment]
This logic may move to the process of `LocalVolume` and `LocalVolumeGroup` queue.

File: pkg/local-storage/member/controller/volume_task_worker.go [hunk 2]
```diff
@@ -258,3 +262,21 @@ func isVolumeReplicaUp(replica *apisv1alpha1.LocalVolumeReplica) bool {
...
+func (m *manager) updateAccessibilityNodes(vol *apisv1alpha1.LocalVolume) error {
+	vol.UpdateAccessibilityNodesFromReplicas()
+	ctx := context.TODO()
+
+	lvg, err := m.queryLocalVolumeGroup(ctx, vol.Spec.VolumeGroup)
+	if err != nil {
+		return err
+	}
+
+	lvg.SetAccessibilityNodes(vol.Spec.Accessibility.Nodes)
```
Comment (SSmallMonster) [inline pkg/local-storage/member/controller/volume_task_worker.go:276]: handle this in the queue of `VolumeGroupTask`

File: pkg/local-storage/member/controller/manager.go [hunk 3]
```diff
@@ -243,6 +244,20 @@ func (m *manager) handleK8sNodeUpdatedEvent(oldObj, newObj interface{}) {
 	}
 }
 
+func (m *manager) handleVolumeCRDUpdateEvent(oldObj, newObj interface{}) {
+	oldVol := oldObj.(*apisv1alpha1.LocalVolume)
+	newVol := newObj.(*apisv1alpha1.LocalVolume)
+
+	// if volume's replica update, we should notify its group
+	if !reflect.DeepEqual(oldVol.Spec.Accessibility.Nodes, newVol.Spec.Accessibility.Nodes) {
```
Comment (SSmallMonster) [inline pkg/local-storage/member/controller/manager.go:252]: what will happen if nodes in accessibility array are same but in a different order？

File: pkg/local-storage/member/controller/volumegroup/manager.go [hunk 4]
```diff
@@ -378,11 +395,53 @@ func (m *manager) processLocalVolumeGroup(lvgName string) error {
...
+			nLvg := lvg.DeepCopy()
+			nLvg.SetAccessibilityNodes(nodes)
+			patch := client.MergeFrom(lvg)
+			if err = m.apiClient.Patch(context.TODO(), nLvg, patch); err != nil {
+				return err
+			}
+		}
+	}
+
 	return m.addLocalVolumeGroup(lvg)
```
Comment (SSmallMonster) [inline pkg/local-storage/member/controller/volumegroup/manager.go:441]: these codes can be combined into a single function `m.addLocalVolumeGroup` and update cache after found a new PVC(volume) in LocalVolumeGroup.

**Comment by sun7927** [comment]
LGTM

**Pull request merged**